packages/core/src/FileUploader/DropZone/DropZone.tsx

```diff
@@ -118,31 +138,32 @@ export const HvDropZone = (props: HvDropZoneProps) => {
 
   const { classes, cx } = useClasses(classesProp);
 
-  const [dragState, setDrag] = useState<boolean>(false);
+  const [dragState, setDragState] = useState(false);
 
   const inputRef = useRef<HTMLInputElement | null>(null);
 
-  const leaveDropArea = () => {
-    setDrag(false);
+  const handleDragLeave = () => {
+    setDragState(false);
   };
 
-  const enterDropArea = () => {
-    setDrag(true);
+  const handleDragEnter: React.DragEventHandler = (event) => {
+    if (disabled) return;
+    event.stopPropagation();
+    event.preventDefault();
+    setDragState(true);
   };
 
   const onChangeHandler = (filesList: FileList) => {
     const filesToProcess = Object.values(filesList);
 
     const newFiles = filesToProcess.map((file) => {
-      const newFile: HvFileData = file;
+      const newFile: HvFileData = {
+        ...file,
```

Comment:
It seems like we can't spread the `file` to copy it, otherwise we get this:

![Screenshot 2024-01-08 at 17 53 45](https://github.com/lumada-design/hv-uikit-react/assets/43220251/292c80ec-4496-4380-8e82-84679f3ccfe3)

We could do something like: 

```tsx
const newFile: HvFileData = file;
newFile.id = uniqueId("uploaded-file-data-");
```

Reply:
Nice catch - we can't copy files with spread 🤦

I ended up still cloning the file, so that we're not mutating the input (and issues with test)